docs/src/start/logging-packets.md

```diff
@@ -0,0 +1,311 @@
+The data structure that we'll need to send information to user-space will need to hold an IPv4 address and an action for Permit/Deny, we'll encode both as a `u32`.
+
+Let's go ahead and add that to `myapp-common/src/lib.rs`
+
+```rust,ignore
+#[repr(C)]
+pub struct PacketLog {
+    pub ipv4_address: u32,
+    pub action: u32,
+}
+
+#[cfg(feature = "user")]
+unsafe impl aya::Pod for PacketLog {}
+```
+
+> 💡 **HINT: Struct Alignment**
+>
+> Structs must be aligned to 8 byte boundaries. You can do this manually, or alternatively you may use `#[repr(packed)]`. If you do not do this, the eBPF verifier will get upset and emit an `invalid indirect read from stack` error.
+
+We implement the `aya::Pod` trait for our struct since it is Plain Old Data as can be safely converted to a byte-slice and back. While not currently required in this example, it is still considered best practice to do so.
+
+### eBPF: Map Creation
+
+Let's create a map called `EVENTS` in `myapp-ebpf/src/main.rs`
+
+```rust,ignore
+use aya_bpf::macros::map;
+use aya_bpf::maps::PerfMap;
+use myapp-common::PacketLog;
+
+#[map(name = "EVENTS")]
+static mut EVENTS: PerfMap<PacketLog> = PerfMap::<PacketLog>::with_max_entries(1024, 0);
+```
+
+When the eBPF program is loaded by Aya, the map will be created for us.
+
+### Userspace: Map Creation
+
+After our call to `probe.attach()` we'll add the following code.
+
+```rust,ignore
+use aya::maps::AsyncPerfEventArray;
+
+let mut perf_array = AsyncPerfEventArray::try_from(bpf.map_mut("EVENTS")?)?;
+```
+
+Our `perf_array` is a mutable reference to the map that was created after the XDP program was loaded by Aya.
+
+## Writing Data
+
+Now we've got our maps set up, let's add some data!
+
+### Generating Bindings To vmlinux.h
+
+To get useful data to add to our maps, we first need some useful data structures to populate with data from the `XdpContext`.
+We want to log the Source IP Address of incoming traffic, so we'll need to:
+
+1. Read the Ethernet Header to determine if this is an IPv4 Packet
+1. Read the Source IP Address from the IPv4 Header
+
+The two structs in the kernel for this are `ethhdr` from `uapi/linux/if_ether.h` and `iphdr` from `uapi/linux/ip.h`.
+If I were to use bindgen to generate Rust bindings for those headers, I'd be tied to the kernel version of the system that I'm developing on.
+This is where `aya-gen` comes in to play. It can easily generate bindings for using the BTF information in `/sys/kernel/btf/vmlinux`.
+
+Once the bindings are generated and checked in to our repository they shouldn't need to be regenerated again unless we need to add a new struct.
+
+Lets use `xtask` to automate this so we can easily reproduce this file in future.
+
+We'll add the following content to `xtask/src/codegen.rs`
+
+```rust,ignore
+use aya_gen::btf_types;
+use std::{
+    fs::File,
+    io::Write,
+    path::{Path, PathBuf},
+};
+
+pub fn generate() -> Result<(), anyhow::Error> {
+    let dir = PathBuf::from("myapp-ebpf/src");
```

Comment:
It would be nice to mention that this path needs to be adjusted if a name other than `myapp` was chosen when the project was generated from the template. Maybe in a paragraph further down, as in "If this prints `No such file or directory (os error 2)`, make sure you have adjusted [...]"